js/bcex.js

```diff
@@ -0,0 +1,397 @@
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let marketId = this.marketId (symbol);
+        let request = {
+            'symbol': marketId,
+        };
+        let response = await this.privatePostApiOrderDepth (this.extend (request, params));
+        let data = response['data'];
+        let orderbook = this.parseOrderBook (data, data['date']);
+        return orderbook;
+    }
+
+    parseMyTrade (trade, market) {
+        let timestamp = trade['created'] * 1000;
+        return {
+            'id': trade['order_id'],
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market.symbol,
+            'type': undefined,
+            'side': trade['side'],
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'number'),
+            'order': undefined,
+            'fee': undefined,
+        };
+    }
+
+    parseMyTrades (trades, market = undefined, since = undefined, limit = undefined) {
+        let result = Object.values (trades || []).map (trade => this.parseMyTrade (trade, market));
+        result = sortBy (result, 'timestamp');
+        let symbol = (typeof market !== 'undefined') ? market['symbol'] : undefined;
+        return this.filterBySymbolSinceLimit (result, symbol, since, limit);
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let marketId = this.marketId (symbol);
+        let request = {
+            'symbol': marketId,
+        };
+        let response = await this.privatePostApiOrderOrderList (this.extend (request, params));
+        let market = this.markets_by_id[marketId];
+        let trades = response['data'];
+        let result = this.parseMyTrades (trades, market);
+        return result;
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'symbol': this.marketId (symbol),
+            'trust_id': id,
+        };
+        let response = await this.privatePostApiOrderOrderInfo (this.extend (request, params));
+        let order = response['data'];
+        let timestamp = order['created'] * 1000;
+        let result = {
+            'info': order,
+            'id': id,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'lastTradeTimestamp': undefined,
+            'symbol': symbol,
+            'type': order['flag'],
+            'side': undefined,
+            'price': order['price'],
+            'cost': undefined,
+            'average': undefined,
+            'amount': order['number'],
+            'filled': order['numberdeal'],
+            'remaining': order['numberover'],
+            'status': order['status'],
```

Comment:
Here status is a number code from: `[Status 0: Not filled 1: Part of transaction 2: All transactions 3: Cancellation of commission]`.

Is there a common format for CCXT that I should convert these to?

Reply:
Yep:

- `open` for non-filled orders
- `open` for partially filled orders
- `closed` for fulfilled orders
- `canceled` for canceled orders see